Brandonle/depositcommand

ble86 opened the pull request:
Added commands .paydeposit and .returndeposit to keep track on master treasurer sheet who in which project has paid their deposit and who has gotten it returned (at the end of the year). In addition to this, added extra code to checkoff command so that checkoffs are processed on the master treasurer sheet to easily determine who gets what amount of their deposit back. Added extra code to addassign so that addassign commands are also kept track of on the master treasurer sheet. An example of the master treasurer sheet will be messaged to Bryan (reviewer) to refer to the format of the sheet. 

Additionally updated the creds.py with current project leads' master roster spreadsheet links, fixed a bug due to inconsistent names of officers between creds.py and lab hours spreadsheet, and changed get_name_from_args definition and call to enforce use of "Jay Park" for name in order to allow new treasurer code to work.

ble86 committed:
Adding deposit sheet command instructions

ble86 committed:
Finished all work on deposit commands

ble86 committed:
Added code to returndeposit, paydeposit, and checkoff to search for the nearest match names for name argument that is incomplete

ble86 committed:
Fixed bot.py and creds.py bug for checking off assignments

ble86 committed:
Undid change to creds.py that was wrong

bryanjwong commented on bot.py:
```diff
@@ -270,6 +276,86 @@ async def labhours(ctx, *args):
+            # Set msg variable to be the embed
+            msg = embed
+
+        # Check if it's a string or embed
```
You can just move the 2 different `await ctx.send()` calls to the if-else block above. Generally we want to avoid vars from having multiple types.

bryanjwong commented on bot.py:
```diff
@@ -270,6 +276,86 @@ async def labhours(ctx, *args):
+            # Set msg variable to be the embed
+            msg = embed
+
+        # Check if it's a string or embed
```
Same as above comment.

bryanjwong commented on bot.py:
```diff
@@ -315,6 +404,30 @@ async def checkoff(ctx, *args):
         else:
             await ctx.send(embed=embed)
 
+        # Get name enforcing use of "Jay Park"
```
I'd move this into a `checkoff_treasurer_sheet` subroutine or something like that for modularity. There's some code duplication where we're calling `get_name_from_args` in the above if block as well, so you can move that outside and pass to the subroutine as a parameter.

bryanjwong commented on bot.py:
```diff
@@ -315,6 +404,30 @@ async def checkoff(ctx, *args):
         else:
```
Whoops, realizing now this multi-typing antipattern was probably added when embeds were implemented... Would you mind cleaning up other occurrences of this?

bryanjwong commented on bot.py:
```diff
@@ -339,8 +452,21 @@ async def addassign(ctx, *args):
+
         await ctx.send(msg)
 
+    # Add assignment to treasurer sheet
```
Again, I'd suggest modularizing this :)

bryanjwong commented on bot.py:
```diff
@@ -515,10 +641,13 @@ def is_officer(ctx):
 # If we do .status OPS Joe Schmoe, this just captures Joe Schmoe as its own string
 # No need to use quotes anymore with this
+
+# Changed this to enforce the use of "Jay Park"
```
Can remove all occurrences of this comment.

Also, could you explain why we are no longer able to build names using `*args`? It's a pretty nice feature and makes running commands a lot more intuitive without needing to capture strings inside `""`.

bryanjwong commented on sheet_transformer.py:
```diff
@@ -106,7 +106,12 @@ def checkoff(self, spread_id, assignment, name, val='y', sheet_index=0):
 
         assignment_col = None
-        for col, a in enumerate(values[0]):
+        # Change row that we're looking for the assignment based off treasurer sheet or project sheet
```
More pythonic approach:
`assignment_row = ASSIGNMENT_ROW if spread_id == TREASURER_SHEET else 0`

bryanjwong commented on sheet_transformer.py:
```diff
@@ -124,6 +129,118 @@ def checkoff(self, spread_id, assignment, name, val='y', sheet_index=0):
+        member_row = None
+
+        # If have same first name, consider them near match
+        near_matches = []
```
Introduces code duplication with `lookup()` routine. I suggest modularizing this near matches behavior.

bryanjwong commented on sheet_transformer.py:
```diff
@@ -124,6 +129,118 @@ def checkoff(self, spread_id, assignment, name, val='y', sheet_index=0):
+        member_row = None
+
+        # If have same first name, consider them near match
+        near_matches = []
```
Same comment as above.

bryanjwong reviewed (commented):
Nice job! Left a few comments. Mostly centered around code cleanliness/keeping code as concise as possible. Also, I might be missing something but why does the new Treasurer sheet break names?

ble86 committed:
Made changes to bot.py and sheet_transformer.py to modularize code according to Bryan's review comments

bryanjwong committed:
change slicing to indexing

bryanjwong committed:
clean up lab hours reminder styling

bryanjwong reviewed (approved):
@ble86 looks good! assuming you've tested everything and it works, feel free to merge 🚀

bryanjwong committed:
revert: slicing to pass array of strings for names

If we index `args`, the `get_name_from_args()` func treats input as array of chars. Instead, use slicing so the func interprets as array of strings.
